### PR TITLE
Vd/documentation

### PR DESCRIPTION
**Pull request by vdebauche**
Add documentation + description folder utils + name project.toml

### CODE REVIEW

**Comment by vdebauche** [comment]
Where can I see the documentation that has been generated ?

File: docs/src/reference/Optim.md [hunk 1]
```diff
@@ -1,14 +1,14 @@
 # Optim 
 
-This folder contains all the different solvers that can be used.
+This folder contains all the different (abstraction-based or not) solvers that can be used. Note that all the solvers are defined using the MathOptInterface framework: for each solver, we define the structure [`AbstractOptimizer`](https://jump.dev/MathOptInterface.jl/stable/reference/models/#MathOptInterface.AbstractOptimizer) and implement the [`Optimize!`](https://jump.dev/MathOptInterface.jl/stable/reference/models/#MathOptInterface.optimize!) function.
```
Comment (adrienbanse) [inline docs/src/reference/Optim.md:3]: It's a subclass of `AbstractOptimizer` but not an object of this type

File: docs/src/reference/System.md [hunk 2]
```diff
@@ -3,23 +3,36 @@
 This folder contains different ways to define systems, for instance to encode a controller.
 
 ## Control system
+
+Each control system should be implemented as a ControlSystem .
```
Comment (adrienbanse) [inline docs/src/reference/System.md:7]: Space before dot

File: docs/src/reference/System.md [hunk 3]
```diff
@@ -3,23 +3,36 @@
...
+
 ```@docs
+Dionysos.System.ControlSystem
+```
+
+So far, we have implemented a few examples of control systems :
```
Comment (adrienbanse) [inline docs/src/reference/System.md:13]: Space before :

File: src/domain/domain_list.jl [hunk 4]
```diff
@@ -7,7 +7,7 @@ end
 """
     DomainList{N,T,S<:Grid{N,T}}
 
-Struct for a basic domain based on a `Grid`
+Struct for a basic domain based on a `Grid` .
```
Comment (adrienbanse) [inline src/domain/domain_list.jl:10]: Space before dot

File: src/domain/grid.jl [hunk 5]
```diff
@@ -113,7 +113,7 @@ end
     GridEllipsoidalRectangular{N,T} <: Grid{N,T}
 
 Uniform grid on rectangular space `rect`, centered at `orig` and with steps set by the vector `h`.
-Cells are (possibly overlapping) ellipsoids defined at each grid point `c` as `(x-c)'P(x-c) ≤ 1`
+Cells are (possibly overlapping) ellipsoids defined at each grid point `c` as `(x-c)'P(x-c) ≤ 1` .
```
Comment (adrienbanse) [inline src/domain/grid.jl:116]: Space before dot

File: src/optim/abstraction/ellipsoids_abstraction.jl [hunk 6]
```diff
@@ -16,7 +16,7 @@ const PR = DI.Problem
 """
     Optimizer{T} <: MOI.AbstractOptimizer
 
-Ellispoids abtraction solver
+Abstraction-based solver for which the domain is covered with elippsoidal cells, independently of the control task.
```
Comment (adrienbanse) [inline src/optim/abstraction/ellipsoids_abstraction.jl:19]: Two p's but one l at eLiPPsoidal

File: src/optim/bemporad_morari.jl [hunk 7]
```diff
@@ -20,7 +20,7 @@ using FillArrays, MathematicalSystems, HybridSystems, JuMP, SemialgebraicSets, P
 """
     Optimizer{T} <: MOI.AbstractOptimizer
 
-Bemporad Morari solver
+Bemporad Morari solver : Optimal control of hybrid systems via a predictive control scheme using mixed integer quadratic programming (MIQP) online optimization procedures.
```
Comment (adrienbanse) [inline src/optim/bemporad_morari.jl:23]: Space before :

File: src/optim/branch_and_bound.jl [hunk 8]
```diff
@@ -16,7 +16,7 @@ using HybridSystems
 """
     Optimizer{T} <: MOI.AbstractOptimizer
 
-Branch and bound solver
+Branch and bound solver : Optimal control of hybrid systems via a predictive control scheme combining a branch and bound algorithm that can refine Q-functions using Lagrangian duality.
```
Comment (adrienbanse) [inline src/optim/branch_and_bound.jl:19]: Space before :

File: src/system/controlsystem.jl [hunk 9]
```diff
@@ -1,4 +1,10 @@
+"""
+The structure
+
+    ControlSystem{N, T}
 
+is the abstract type which defines a control system.
```
Comment (adrienbanse) [inline src/system/controlsystem.jl:6]: "that" instead of "which"?

File: src/system/controlsystem.jl [hunk 10]
```diff
@@ -164,7 +170,7 @@ end
 """
     SimpleSystem{N, T, F <: Function, F2} <: ControlSystem{N, T}   
 
-TO ADD
+is one implementation of the `ControlSystem` type
```
Comment (adrienbanse) [inline src/system/controlsystem.jl:173]: dot missing

File: src/system/trajectory.jl [hunk 11]
```diff
@@ -10,7 +10,7 @@ using Polyhedra
     DiscreteTrajectory{Q, TT}
 
 `q_0` is the starting mode and `transitions` is a sequence of discrete
-transitions in the system
+transitions in the system.
```
Comment (adrienbanse) [inline src/system/trajectory.jl:13]: Remove return?

File: src/system/trajectory.jl [hunk 12]
```diff
@@ -37,7 +37,7 @@ end
     ContinuousTrajectory{T, XVT<:AbstractVector{T}, UVT<:AbstractVector{T}}
 
 `x` is a sequence of points in the state space and `u` is a sequence of points
-in the input space
+in the input space.
```
Comment (adrienbanse) [inline src/system/trajectory.jl:40]: Remove return?

**Review by adrienbanse — COMMENTED** [review]
Thanks a lot @vdebauche, crazy job

**Review by adrienbanse — APPROVED** [review]

**Pull request merged**